wrapper/src/main/java/software/amazon/jdbc/states/SessionStateHelper.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.states;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.EnumSet;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class SessionStateHelper {
+
+  /**
+   * Transfers basic session state from one connection to another.
+   *
+   * @param sessionState Session state of from-connection
+   * @param from The connection to transfer state from
+   * @param to   The connection to transfer state to
+   * @throws SQLException if a database access error occurs, this method is called on a closed connection, this
+   *                      method is called during a distributed transaction, or this method is called during a
+   *                      transaction
+   */
+  public void transferSessionState(
```

Comment:
NIT: wondering if it wouldn't be clearer to have `src` and `dest` instead of `from` and `to` (ex: `dest.setReadOnly(src.isReadOnly());`)